internal/services/bot/bot_channel_email_resource.go

```diff
@@ -101,6 +109,17 @@ func resourceBotChannelEmailCreate(d *pluginsdk.ResourceData, meta interface{})
 		Kind:     botservice.KindBot,
 	}
 
+	if v, ok := d.GetOk("email_password"); ok {
+		channel, _ := channel.Properties.AsEmailChannel()
+		channel.Properties.Password = utils.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("magic_code"); ok {
+		channel, _ := channel.Properties.AsEmailChannel()
+		channel.Properties.AuthMethod = utils.Float(1)
```

Comment:
@tombuildsstuff , maybe it's by design since they added the meaning of the value. "0" means "Password" by default. "1" means "OAuth". See more details from [rest api spec](https://github.com/Azure/azure-rest-api-specs/blob/80c21c17b4a7aa57f637ee594f7cfd653255a7e0/specification/botservice/resource-manager/Microsoft.BotService/stable/2022-09-15/botservice.json#L2194). Does it make sense?

Reply:
Service team confirmed that it's by design to them.

Reply:
@neil-yechenwei - then there should be a swagger bug raised to add these constants? as well if it 0 and 1 then change the type from float to in?

Reply:
Updated PR for this concern.